### PR TITLE
Added ssl support for redis

### PR DESCRIPTION
**Pull request by dashevchenko**
## Pull Request description

Added ssl support for redis

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/resources/thingsboard.yml [hunk 1]
```diff
@@ -639,6 +639,36 @@ redis:
   db: "${REDIS_DB:0}"
   # db password
   password: "${REDIS_PASSWORD:}"
+  # ssl config
+  ssl:
+    # Enable/disable secure connection
+    enabled: "${REDIS_SSL_ENABLED:false}"
```
Comment (YevhenBondarenko) [inline application/src/main/resources/thingsboard.yml:645]: In other properties below, used TB_REDIS....
Plz use TB_REDIS_SSL_ENABLED (for each yml file)

File: common/cache/src/main/java/org/thingsboard/server/cache/TBRedisStandaloneConfiguration.java [hunk 2]
```diff
@@ -57,32 +57,36 @@ public class TBRedisStandaloneConfiguration extends TBRedisCacheConfiguration {
...
+                    .sslSocketFactory(createSslSocketFactory());
+        }
+        if (usePoolConfig) {
+            jedisClientConfigurationBuilder
+                    .usePooling()
+                    .poolConfig(buildPoolConfig());
         }
+        return jedisClientConfigurationBuilder.build();
     }
 }
```
Comment (YevhenBondarenko) [inline common/cache/src/main/java/org/thingsboard/server/cache/TBRedisStandaloneConfiguration.java:92]: Add empty line

**Review by YevhenBondarenko — COMMENTED** [review]
Plz compare createSslSocketFactory() from TBRedisCacheConfiguration with SslCredentialsConfig (and entities like `RedisSslCredentialsConfiguration`) I think the main logic for creating SSL factory can be similar, and you can refactor and reuse the same code in both this SSL configuration?

**Review by YevhenBondarenko — APPROVED** [review]

File: application/src/main/resources/thingsboard.yml [hunk 3]
```diff
@@ -639,6 +639,36 @@ redis:
   db: "${REDIS_DB:0}"
   # db password
   password: "${REDIS_PASSWORD:}"
+  # ssl config
+  ssl:
+    # Enable/disable secure connection
+    enabled: "${TB_REDIS_SSL_ENABLED:false}"
+    # Server SSL credentials
+    credentials:
+      # Server credentials type (PEM - pem certificate file; KEYSTORE - java keystore)
```
Comment (ViacheslavKlimov) [inline application/src/main/resources/thingsboard.yml:648]: Either change description and use pem and keystore in lowercase, or use `equalsIgnoreCase` in `createAndInitTrustManagerFactory`.

**Review by ViacheslavKlimov — APPROVED** [review]

**Pull request merged**